site/components/Input.js

```diff
@@ -26,9 +28,28 @@ const SuffixedInput = ({ suffix, ...props }) => (
 
 const Caption = ({ caption, captionColor }) =>
   caption && (
-    <p className={`text-center text-sm mt-2 ${captionColor}`}>{caption}</p>
+    <div className="absolute flex gap-1 items-center">
+      <CaptionIcon captionColor={captionColor} />
+      <p className={` text-sm pr-4 ${captionColor}`}>{caption}</p>
+    </div>
   )
 
+const CaptionIcon = ({ captionColor }) => {
+  const [name, setName] = useState('')
+
+  useEffect(() => {
+    if (captionColor.includes('red')) {
+      setName('error')
+    }
+
+    if (captionColor.includes('green')) {
+      setName('check')
+    }
+  }, [])
+
+  return <SvgContainer className={`w-6 h-6`} name={name} />
+}
+
```

Comment:
I think `useState` and `useEffect` are not needed here. You can rewrite this as just deriving values from props:

```suggestion
const CaptionIcon = ({ captionColor }) => {
  const name = captionColor.includes('red')
       ? 'error'
       : 
          captionColor.includes('green')
             ? 'check'
             : ''

  return <SvgContainer className={`w-6 h-6`} name={name} />
}
```

you can rewrite this to use i/else or whatever you prefer, but the point is that both the state and the effect are not needed.

Reply:
done